src/vs/platform/environment/node/argv.ts

```diff
@@ -217,6 +220,39 @@ export function buildVersionMessage(version: string | undefined, commit: string
 	return `${version || localize('unknownVersion', "Unknown version")}\n${commit || localize('unknownCommit', "Unknown commit")}\n${process.arch}`;
 }
 
+export function buildTelemetryMessage(appRoot: string, extensionsPath: string): string {
+	// const contents = readFileSync('./telemetry.json');
+	// Gets all the directories inside the extension directory
+	const dirs = readdirSync(extensionsPath).filter(files => statSync(join(extensionsPath, files)).isDirectory());
+	const telemetryJsonFolders: string[] = [];
+	dirs.forEach((dir) => {
+		const files = readdirSync(join(extensionsPath, dir)).filter(file => file === 'telemetry.json');
+		// We know it contains a telemetry.json file so we add it to the list of folders which have one
+		if (files.length === 1) {
+			telemetryJsonFolders.push(dir);
+		}
+	});
+	const mergedTelemetry = Object.create(null);
+	// Simple function to merge the telemetry into one json object
+	const mergeTelemetry = (contents: string, dirName: string) => {
+		const telemetryData = JSON.parse(contents);
+		mergedTelemetry[dirName] = telemetryData;
+	};
+	telemetryJsonFolders.forEach((folder) => {
+		const contents = readFileSync(join(extensionsPath, folder, 'telemetry.json')).toString();
+		mergeTelemetry(contents, folder);
+	});
+	try {
+		let contents = readFileSync(path.join(appRoot, 'out/', 'telemetry-core.json')).toString();
```

Comment:
don't use path from node. use our implementation of join which is already imported into this file